src/pixelflow/core.py

```diff
@@ -102,10 +110,49 @@ def pixelflow(
         * Image padding to take care of edges
         * Supplying different image channels to different functions
     """
-    features = regionprops_table(
-        label(mask),
-        image,
-        properties=features,
-        extra_properties=custom,
-    )
-    return PixelflowResult(features=pd.DataFrame(features))
+
+    if dim_labels == "":
+        if mask.ndim == 2:
+            dim_labels = "YX"
+            print("YX image detected")
+        elif mask.ndim == 3:
```

Comment:
I'll just flag this for later, but there are situations where the mask dimensions and the image dimensions may not match. For example, a 2D mask with a 3D image (YXC) where the same mask is used for each C

Reply:
Agreed. That was the reason I used the mask rather than the image dimensions for the moment to estimate the dimensions. When the code does something more useful with YXC images then I'm assuming this will be updated.